src/Security/AnonymousUserAuthenticator.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Buddy\Repman\Security;
+
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Core\Exception\BadCredentialsException;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+use Symfony\Component\Security\Guard\AbstractGuardAuthenticator;
+
+final class AnonymousUserAuthenticator extends AbstractGuardAuthenticator
+{
+    /**
+     * @codeCoverageIgnore
+     *
+     * @return Response
+     */
+    public function start(Request $request, AuthenticationException $authException = null)
+    {
+        return new JsonResponse([
+            'message' => 'Authentication Required',
+        ], Response::HTTP_UNAUTHORIZED);
+    }
+
+    public function supports(Request $request)
+    {
+        return $request->get('_route') !== 'repo_package_downloads'
+            && !$request->headers->has('PHP_AUTH_USER')
+            && !$request->headers->has('PHP_AUTH_PW');
+    }
+
+    public function getCredentials(Request $request)
+    {
+        $organizationAlias = $request->get('organization');
+        if ($organizationAlias === null) {
+            throw new BadCredentialsException();
+        }
+
+        return $organizationAlias;
+    }
+
+    public function getUser($credentials, UserProviderInterface $userProvider)
+    {
+        return $userProvider->loadUserByUsername('alias:'.$credentials);
```

Comment:
why not check if `$userProvider` is `OrganizationProvider` and use `loadUserByAlias` method (as public). This will allow to remove this `alias:` magic string.

Reply:
That was also my first thought, but it is not a part of `UserProviderInterface` so phpstan will be unhappy about undeclared method

Reply:
`instance of` will help with that